src/rust/dns/lib.rs

```diff
@@ -0,0 +1,174 @@
+#[cxx::bridge(namespace = "workerd::rust::dns")]
+mod ffi {
+    /// CAA record representation
+    struct CaaRecord {
+        critical: u8,
+        field: String,
+        value: String,
+    }
+    /// NAPTR record representation
+    struct NaptrRecord {
+        flags: String,
+        service: String,
+        regexp: String,
+        replacement: String,
+        order: u32,
+        preference: u32,
+    }
+    extern "Rust" {
+        fn parse_caa_record(record: &str) -> CaaRecord;
```

Comment:
these should be Result. Parsing could fail by definition. Result is supported by cxx and is transformed into exception.